clang/lib/Driver/Driver.cpp

```diff
@@ -5527,9 +5527,11 @@ const char *Driver::GetNamedOutputPath(Compilation &C, const JobAction &JA,
   if ((!AtTopLevel && !isSaveTempsEnabled() &&
        (!C.getArgs().hasArg(options::OPT__SLASH_Fo) ||
         // FIXME - The use of /Fo is limited when offloading is enabled.  When
-        // compiling to exe use of /Fo does not produce the named obj
+        // compiling to exe use of /Fo does not produce the named obj.  We also
+        // should not use the named output when performing unbundling.
         (C.getArgs().hasArg(options::OPT__SLASH_Fo) &&
          (!JA.isOffloading(Action::OFK_None) ||
+          isa<OffloadUnbundlingJobAction>(JA) ||
```

Comment:
Would a `false`-returning virtual overload of `isOffloading()` for `OffloadUnbundlingJobAction` be a tad cleaner? Although, I'm not sure if making `isOffloading()` virtual in the `Action` class would be appropriate in the first place.

Reply:
I don't know if it would be appropriate either.  The only Action that would take advantage of the overload would be the <code>OffloadUnbundlingJobAction</code> due to the fact that it is basically the only action that is a host based action only.  Unbundling does not occur on the offload toolchain.

Reply:
Right, that's reasonable.